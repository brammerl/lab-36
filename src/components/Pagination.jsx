import React from 'react';
import usePagination from '../../hooks/appContext';

const Pagination = () => {
  const page = usePagination();

