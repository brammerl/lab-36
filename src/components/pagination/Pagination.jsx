import React from 'react';
import { useDispatch } from '../../hooks/appContext';

const Pagination = ({page, dispatch}) => {
  
  const increment = () => dispatch({type: 'ADD_PAGE'});
  const decrement = () => dispatch({type: 'SUB_PAGE'});

  return (
    <>
      Page: {page}
      <button onClick={decrement}>Prev</button>
      <button onClick={increment}>Next</button>
    </>
  )
};

export default Pagination;
