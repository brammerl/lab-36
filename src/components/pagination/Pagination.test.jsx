import React from 'react';
import { shallow } from 'enzyme';
import Pagination from './Pagination';

describe('Pagination component', () => {
  it('renders Pagination', () => {
    const wrapper = shallow(<Pagination />);
    expect(wrapper).toMatchSnapshot();
  });
});
  