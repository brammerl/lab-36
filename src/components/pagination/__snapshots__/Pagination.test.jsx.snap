// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component renders Pagination 1`] = `
<Fragment>
  Page: 
  <button
    onClick={[Function]}
  >
    Prev
  </button>
  <button
    onClick={[Function]}
  >
    Next
  </button>
</Fragment>
`;
